cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/DefaultScheduleBuilder.java

```diff
@@ -118,4 +118,11 @@ public ScheduleBuilder waitUntilMet() {
     constraints.get(constraints.size() - 1).setWaitUntilMet(true);
     return this;
   }
+
+  @Override
+  public ScheduleBuilder abortIfNotMet() {
+    // user will only be able to call abortIfNotMet right after they add a Constraint
+    constraints.get(constraints.size() - 1).setWaitUntilMet(false);
+    return this;
```

Comment:
so I can do this: 
```
...withTimeWindow("13:12", "11:10").abortIfNotMet().waitUntilMet().setProperties(...).abortIfNotMet()...
```

Not a biggie, but perhaps it's better to avoid that?